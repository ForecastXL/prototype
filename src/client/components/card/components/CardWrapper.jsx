import PropTypes from 'prop-types'
import { Card as MuiCard } from 'material-ui/Card'
import styled from 'styled-components'

// Determine the width based on either a fixed width or a relative size indicator.
const determineCardWidth = ({ width, size }) => {
  let cssWidth = '100%'
  if (width) {
    cssWidth = width
  } else if (size) {
    // Note: these should change based on the viewport (should be responsive)
    switch (size) {
      case 'small':
        cssWidth = '40rem'
        break
      case 'medium':
        cssWidth = '60rem'
        break
      case 'large':
        cssWidth = '90rem'
        break
      default:
        throw new Error(`Unknown size '${size}'`)
    }
  }
  return cssWidth
}

const CardWrapper = styled(MuiCard)`
  margin: ${props => props.marginTop} auto 0;
  padding: ${props => props.padding};
  width: ${props => determineCardWidth(props)};
`

CardWrapper.propTypes = {
  marginTop: PropTypes.string,
  padding: PropTypes.string,
  width: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large'])
}

CardWrapper.defaultProps = {
  marginTop: '5rem',
  padding: '1rem'
}

export default CardWrapper
