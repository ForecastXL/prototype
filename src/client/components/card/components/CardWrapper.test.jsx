/**
 * @jest-environment node
 */

import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import CardWrapper from './CardWrapper'

jest.mock('material-ui/Card', () => ({
  Card: () => <div />
}))

describe('<CardWrapper />', () => {
  it('renders a card wrapper at 100% width by default', () => {
    const wrapper = shallow(<CardWrapper />)
    expect(shallowToJson(wrapper)).toMatchStyledComponentsSnapshot()
  })

  it('renders a card wrapper at small size when given size = small', () => {
    const wrapper = shallow(<CardWrapper size="small" />)
    expect(shallowToJson(wrapper)).toMatchStyledComponentsSnapshot()
  })

  it('renders a card wrapper at medium size when given size = medium', () => {
    const wrapper = shallow(<CardWrapper size="medium" />)
    expect(shallowToJson(wrapper)).toMatchStyledComponentsSnapshot()
  })

  it('renders a card wrapper at large size when given size = large', () => {
    const wrapper = shallow(<CardWrapper size="large" />)
    expect(shallowToJson(wrapper)).toMatchStyledComponentsSnapshot()
  })

  it('renders a card wrapper at a specific width when given a width', () => {
    const wrapper = shallow(<CardWrapper width="20rem" />)
    expect(shallowToJson(wrapper)).toMatchStyledComponentsSnapshot()
  })
})
