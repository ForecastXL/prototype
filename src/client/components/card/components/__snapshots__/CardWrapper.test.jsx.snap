// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardWrapper /> renders a card wrapper at 100% width by default 1`] = `
.fnzPNO {
  margin: 5rem auto 0;
  padding: 1rem;
  width: 100%;
}

<Card
  className="sc-bdVaJa fnzPNO"
  marginTop="5rem"
  padding="1rem"
/>
`;

exports[`<CardWrapper /> renders a card wrapper at a specific width when given a width 1`] = `
.ljQsVH {
  margin: 5rem auto 0;
  padding: 1rem;
  width: 20rem;
}

<Card
  className="sc-bdVaJa ljQsVH"
  marginTop="5rem"
  padding="1rem"
  width="20rem"
/>
`;

exports[`<CardWrapper /> renders a card wrapper at large size when given size = large 1`] = `
.kkckIN {
  margin: 5rem auto 0;
  padding: 1rem;
  width: 90rem;
}

<Card
  className="sc-bdVaJa kkckIN"
  marginTop="5rem"
  padding="1rem"
  size="large"
/>
`;

exports[`<CardWrapper /> renders a card wrapper at medium size when given size = medium 1`] = `
.jyfWop {
  margin: 5rem auto 0;
  padding: 1rem;
  width: 60rem;
}

<Card
  className="sc-bdVaJa jyfWop"
  marginTop="5rem"
  padding="1rem"
  size="medium"
/>
`;

exports[`<CardWrapper /> renders a card wrapper at small size when given size = small 1`] = `
.jWDGKL {
  margin: 5rem auto 0;
  padding: 1rem;
  width: 40rem;
}

<Card
  className="sc-bdVaJa jWDGKL"
  marginTop="5rem"
  padding="1rem"
  size="small"
/>
`;
